diff --git a/inc/config.h b/inc/config.h
index 9f82c66..2dca015 100644
--- a/inc/config.h
+++ b/inc/config.h
@@ -3,8 +3,8 @@
 #define PKG_RDB "/var/pkg/remote/"
 #define PKG_TMP "/var/pkg/cache/"
 
-#define PKG_SRC "url"
-#define PKG_SDB "url"
+#define PKG_SRC "https://sourceforge.net/projects/eltanin-os/files/repository/glacies/x86_64/"
+#define PKG_SDB "https://sourceforge.net/projects/eltanin-os/files/repository/glacies/"
 #define PKG_FDB "remote.db"
 #define PKG_FMT ".pkg.tzz"
 #define PKG_SIG ".sig"
diff --git a/src/lux.c b/src/lux.c
index 54c442c..252a255 100644
--- a/src/lux.c
+++ b/src/lux.c
@@ -204,7 +204,7 @@ fetch(Package *pkg)
 		}
 
 		tmp.n -= membuf_vstrcat(&tmp, PKG_SRC,
-		         pkg->name, "#", pkg->version);
+		         pkg->name, "#", pkg->version, "/download");
 		if (netfd(tmp.p, fd[i], NULL) < 0)
 			goto failure;
 	}
@@ -358,7 +358,7 @@ update(void)
 		goto failure;
 	}
 
-	p.n -= membuf_vstrcat(&p, PKG_SDB, PKG_FDB);
+	p.n -= membuf_vstrcat(&p, PKG_SDB, PKG_FDB, "/download");
 
 	if (netfd(p.p, fd[0], NULL) < 0)
 		goto failure;
