diff -Naur a/src/config.h c/src/config.h
--- a/src/config.h	Wed Jun 22 04:13:58 2011
+++ c/src/config.h	Wed Dec 19 21:48:53 2018
@@ -20,10 +20,9 @@
 #endif
 
 #if !__MSDOS__
-#if !defined(__GLIBC__) || (__GLIBC__ < 2) || \
-	 !defined(__GLIBC_MINOR__) || (__GLIBC_MINOR__ < 1)
-# warning "glibc version 2.1 or later is recommended"
-#endif /* !__MSDOS__ */
+/* assume glibc-2.1 compatible environment */
+#define __GLIBC__       2
+#define __GLIBC_MINOR__ 1
 
 #define TMP_DEV     "/tmp/dev.%d" /* temporary devices are created here */
 #define MAX_TMP_DEV 50 /* highest temp. device number */
diff -Naur a/src/lilo.c c/src/lilo.c
--- a/src/lilo.c	Sat Nov 21 21:51:24 2015
+++ c/src/lilo.c	Wed Dec 19 21:49:42 2018
@@ -182,7 +182,6 @@
                 printf("Without");
 #endif
                 printf(" device-mapper\n");
-		printf("\nglibc version %d.%d\n", __GLIBC__, __GLIBC_MINOR__);
 		printf("Kernel Headers included from  %d.%d.%d\n",
 			LINUX_VERSION_CODE>>16,
 			LINUX_VERSION_CODE>>8 & 255,
