diff --git a/Makefile b/Makefile
index b7488add..089a8b75 100644
--- a/Makefile
+++ b/Makefile
@@ -1,128 +1,194 @@
-# SPDX-License-Identifier: GPL-2.0
-# Top level Makefile for iproute2
-
-ifeq ("$(origin V)", "command line")
-VERBOSE = $(V)
-endif
-ifndef VERBOSE
-VERBOSE = 0
-endif
-
-ifeq ($(VERBOSE),0)
-MAKEFLAGS += --no-print-directory
-endif
-
-PREFIX?=/usr
-LIBDIR?=$(PREFIX)/lib
-SBINDIR?=/sbin
-CONFDIR?=/etc/iproute2
-NETNS_RUN_DIR?=/var/run/netns
-NETNS_ETC_DIR?=/etc/netns
-DATADIR?=$(PREFIX)/share
-HDRDIR?=$(PREFIX)/include/iproute2
-DOCDIR?=$(DATADIR)/doc/iproute2
-MANDIR?=$(DATADIR)/man
-ARPDDIR?=/var/lib/arpd
-KERNEL_INCLUDE?=/usr/include
-BASH_COMPDIR?=$(DATADIR)/bash-completion/completions
-
-# Path to db_185.h include
-DBM_INCLUDE:=$(DESTDIR)/usr/include
-
-SHARED_LIBS = y
-
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
-ifneq ($(SHARED_LIBS),y)
-DEFINES+= -DNO_SHARED_LIBS
-endif
-
-DEFINES+=-DCONFDIR=\"$(CONFDIR)\" \
-         -DNETNS_RUN_DIR=\"$(NETNS_RUN_DIR)\" \
-         -DNETNS_ETC_DIR=\"$(NETNS_ETC_DIR)\"
-
-#options for decnet
-ADDLIB+=dnet_ntop.o dnet_pton.o
-
-#options for ipx
-ADDLIB+=ipx_ntop.o ipx_pton.o
-
-#options for mpls
-ADDLIB+=mpls_ntop.o mpls_pton.o
-
-CC := gcc
-HOSTCC ?= $(CC)
-DEFINES += -D_GNU_SOURCE
-# Turn on transparent support for LFS
-DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CCOPTS = -O2
-WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
-WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
-
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS)
-YACCFLAGS = -d -t -v
-
-SUBDIRS=lib ip tc bridge misc netem genl tipc devlink rdma man
-
-LIBNETLINK=../lib/libutil.a ../lib/libnetlink.a
-LDLIBS += $(LIBNETLINK)
-
-all: config.mk
-	@set -e; \
-	for i in $(SUBDIRS); \
-	do echo; echo $$i; $(MAKE) $(MFLAGS) -C $$i; done
-
-help:
-	@echo "Make Targets:"
-	@echo " all                 - build binaries"
-	@echo " clean               - remove products of build"
-	@echo " distclean           - remove configuration and build"
-	@echo " install             - install binaries on local machine"
-	@echo " check               - run tests"
-	@echo " cscope              - build cscope database"
-	@echo " snapshot            - generate version number header"
-	@echo ""
-	@echo "Make Arguments:"
-	@echo " V=[0|1]             - set build verbosity level"
-
-config.mk:
-	sh configure $(KERNEL_INCLUDE)
-
+.SUFFIXES:
+.SUFFIXES: .o .c
+
+INTERNAL_CFLAGS=\
+	-D_GNU_SOURCE\
+	-DHAVE_SETNS\
+	-Iinclude\
+	-Iinclude/uapi\
+	-Imisc
+
+BIN=\
+	bin/bridge\
+	bin/ip\
+	bin/ss
+
+IPSRC=\
+	ip/ip.c\
+	ip/ip6tunnel.c\
+	ip/ipaddress.c\
+	ip/ipaddrlabel.c\
+	ip/ipfou.c\
+	ip/ipila.c\
+	ip/ipl2tp.c\
+	ip/iplink.c\
+	ip/iplink_bond.c\
+	ip/iplink_bond_slave.c\
+	ip/iplink_bridge.c\
+	ip/iplink_bridge_slave.c\
+	ip/iplink_can.c\
+	ip/iplink_dummy.c\
+	ip/iplink_geneve.c\
+	ip/iplink_hsr.c\
+	ip/iplink_ifb.c\
+	ip/iplink_ipoib.c\
+	ip/iplink_ipvlan.c\
+	ip/iplink_macvlan.c\
+	ip/iplink_netdevsim.c\
+	ip/iplink_nlmon.c\
+	ip/iplink_rmnet.c\
+	ip/iplink_team.c\
+	ip/iplink_vcan.c\
+	ip/iplink_vlan.c\
+	ip/iplink_vrf.c\
+	ip/iplink_vxcan.c\
+	ip/iplink_vxlan.c\
+	ip/iplink_xdp.c\
+	ip/iplink_xstats.c\
+	ip/ipmacsec.c\
+	ip/ipmaddr.c\
+	ip/ipmonitor.c\
+	ip/ipmroute.c\
+	ip/ipneigh.c\
+	ip/ipnetconf.c\
+	ip/ipnetns.c\
+	ip/ipntable.c\
+	ip/ipprefix.c\
+	ip/iproute.c\
+	ip/iproute_lwtunnel.c\
+	ip/iprule.c\
+	ip/ipseg6.c\
+	ip/iptoken.c\
+	ip/iptunnel.c\
+	ip/iptuntap.c\
+	ip/ipvrf.c\
+	ip/ipxfrm.c\
+	ip/link_gre.c\
+	ip/link_gre6.c\
+	ip/link_ip6tnl.c\
+	ip/link_iptnl.c\
+	ip/link_veth.c\
+	ip/link_vti.c\
+	ip/link_vti6.c\
+	ip/rtm_map.c\
+	ip/tcp_metrics.c\
+	ip/tunnel.c\
+	ip/xfrm_monitor.c\
+	ip/xfrm_policy.c\
+	ip/xfrm_state.c
+
+BRIDGESRC=\
+	bridge/bridge.c\
+	bridge/fdb.c\
+	bridge/link.c\
+	bridge/mdb.c\
+	bridge/monitor.c\
+	bridge/vlan.c
+
+SSSRC=\
+	misc/ss.c\
+	misc/ssfilter.tab.c
+
+MAN8=\
+	man/man8/bridge.8\
+	man/man8/ip.8\
+	man/man8/ss.8\
+	man/man8/ip-addrlabel.8\
+	man/man8/ip-fou.8\
+	man/man8/ip-gue.8\
+	man/man8/ip-l2tp.8\
+	man/man8/ip-macsec.8\
+	man/man8/ip-maddress.8\
+	man/man8/ip-monitor.8\
+	man/man8/ip-mroute.8\
+	man/man8/ip-neighbour.8\
+	man/man8/ip-netconf.8\
+	man/man8/ip-netns.8\
+	man/man8/ip-ntable.8\
+	man/man8/ip-rule.8\
+	man/man8/ip-sr.8\
+	man/man8/ip-tcp_metrics.8\
+	man/man8/ip-token.8\
+	man/man8/ip-tunnel.8\
+	man/man8/ip-vrf.8\
+	man/man8/ip-xfrm.8
+
+LIBNETLINKSRC=\
+	lib/bpf.c\
+	lib/color.c\
+	lib/dnet_ntop.c\
+	lib/dnet_pton.c\
+	lib/exec.c\
+	lib/fs.c\
+	lib/inet_proto.c\
+	lib/ipx_ntop.c\
+	lib/ipx_pton.c\
+	lib/json_print.c\
+	lib/json_writer.c\
+	lib/libgenl.c\
+	lib/libnetlink.c\
+	lib/ll_addr.c\
+	lib/ll_map.c\
+	lib/ll_proto.c\
+	lib/ll_types.c\
+	lib/mpls_ntop.c\
+	lib/mpls_pton.c\
+	lib/names.c\
+	lib/namespace.c\
+	lib/rt_names.c\
+	lib/utils.c
+
+LIBNETLINK=    lib/libnetlink.a
+LIBNETLINKOBJ= $(LIBNETLINKSRC:.c=.o)
+
+IPOBJ=$(IPSRC:.c=.o)
+BRIDGEOBJ=$(BRIDGESRC:.c=.o)
+SSOBJ=$(SSSRC:.c=.o)
+
+OBJ= $(BIN:.c=.o) $(IPOBJ) $(BRIDGEOBJ) $(SSOBJ) $(LIBNETLINKOBJ)
+LIB= $(LIBNETLINK)
+
+#
+all: $(BIN) $(LIB)
+
+#
+misc/ssfilter.tab.c: misc/ssfilter.y
+	$(YACC) $(YFLAGS) misc/ssfilter.y
+	mv y.tab.c misc/ssfilter.tab.c
+	mv y.tab.h misc/ssfilter.tab.h
+
+#
+.c.o:
+	$(CC) $(INTERNAL_CFLAGS) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
+
+#
+bin/bridge: $(BRIDGEOBJ) $(LIBNETLINK)
+	@mkdir bin 2>/dev/null || true
+	$(CC) $(LDFLAGS) -o $@ $(BRIDGEOBJ) $(LIB)
+
+bin/ip: $(IPOBJ) $(LIBNETLINK)
+	@mkdir bin 2>/dev/null || true
+	$(CC) $(LDFLAGS) -o $@ $(IPOBJ) $(LIB)
+
+bin/ss: $(SSOBJ) $(LIBNETLINK)
+	@mkdir bin 2>/dev/null || true
+	$(CC) $(LDFLAGS) -o $@ $(SSOBJ) $(LIB)
+
+#
+$(LIBNETLINK): $(LIBNETLINKOBJ)
+	$(AR) rc $@ $?
+	$(RANLIB) $@
+
+#
 install: all
-	install -m 0755 -d $(DESTDIR)$(SBINDIR)
-	install -m 0755 -d $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(ARPDDIR)
-	install -m 0755 -d $(DESTDIR)$(HDRDIR)
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples
-	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples/diffserv
-	install -m 0644 README.iproute2+tc $(shell find examples -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples
-	install -m 0644 $(shell find examples/diffserv -maxdepth 1 -type f) \
-		$(DESTDIR)$(DOCDIR)/examples/diffserv
-	@for i in $(SUBDIRS);  do $(MAKE) -C $$i install; done
-	install -m 0644 $(shell find etc/iproute2 -maxdepth 1 -type f) $(DESTDIR)$(CONFDIR)
-	install -m 0755 -d $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 bash-completion/tc $(DESTDIR)$(BASH_COMPDIR)
-	install -m 0644 include/bpf_elf.h $(DESTDIR)$(HDRDIR)
-
-snapshot:
-	echo "static const char SNAPSHOT[] = \""`date +%y%m%d`"\";" \
-		> include/SNAPSHOT.h
+	install -dm 755        $(DESTDIR)/$(PREFIX)/bin
+	$(STRIP) $(BIN)
+	install -cm 755 $(BIN) $(DESTDIR)/$(PREFIX)/bin
+	install -dm 755         $(DESTDIR)/$(MANDIR)/man8
+	install -cm 644 $(MAN8) $(DESTDIR)/$(MANDIR)/man8

 clean:
-	@for i in $(SUBDIRS) testsuite; \
-	do $(MAKE) $(MFLAGS) -C $$i clean; done
-
-clobber:
-	touch config.mk
-	$(MAKE) $(MFLAGS) clean
-	rm -f config.mk cscope.*
-
-distclean: clobber
-
-check:
-	cd testsuite && $(MAKE) && $(MAKE) alltests
-
-cscope:
-	cscope -b -q -R -Iinclude -sip -slib -smisc -snetem -stc
-
-.EXPORT_ALL_VARIABLES:
+	rm -f $(BIN) $(OBJ) $(LIB)
+	rm -f misc/ssfilter.tab.c
+	rm -f misc/ssfilter.tab.h
+	rmdir bin
