diff -Naur a/make.vars b/make.vars
--- a/make.vars	2013-06-07 04:50:30.000000000 -0300
+++ b/make.vars	2019-01-29 05:42:52.709103342 -0200
@@ -99,9 +99,9 @@
   cat $$DESTDIR/etc/lilo.defines; else echo $(CONFIG); fi ) | \
   sed 's/-D/-DLCF_/g'`
 LILO=-DLILO=0xbb920890
-KVER=`if [ -f /usr/include/linux/version.h ]; then echo -DHAS_VERSION_H; fi`
-HDM=`if [ -f /usr/include/libdevmapper.h ]; then echo -DHAS_LIBDEVMAPPER_H; fi`
-DEVMAPPER=`if [ -f /usr/include/libdevmapper.h ]; then echo "-ldevmapper"; fi`
+KVER="-DHAS_VERSION_H"
+HDM=""
+DEVMAPPER=""
 
 TMP=_TmP_
 
