diff --git a/inc/config.h b/inc/config.h
index 920b6c6..c3c57f8 100644
--- a/inc/config.h
+++ b/inc/config.h
@@ -3,8 +3,8 @@
 #define PKG_RDB "/var/pkg/remote/"
 #define PKG_TMP "/var/pkg/cache/"

-#define PKG_SRC "url"
-#define PKG_SDB "url"
+#define PKG_SRC "https://downloads.sourceforge.net/project/eltanin-os/repository/glacies/x86_64/"
+#define PKG_SDB "https://downloads.sourceforge.net/project/eltanin-os/repository/glacies/"
 #define PKG_FDB "remote.db"
 #define PKG_FMT ".pkg.tzz"
 #define PKG_SIG ".sig"
