diff -Naur a/src/edit.c b/src/edit.c
--- a/src/edit.c	2015-11-21 21:50:17.000000000 -0200
+++ b/src/edit.c	2019-08-12 20:38:58.024217492 -0300
@@ -192,8 +192,7 @@
 	   BITMAPLILOHEADER bmlh;
 	} tm;
 static	MENUTABLE *menu = &tm.mt;
-static	BITMAPLILOHEADER *lh = (void*)(tm.buffer +
-    			((intptr_t)&tm.mt.row - (intptr_t)&tm.bmlh.row));
+static	BITMAPLILOHEADER *lh;
 
 /* a convenience definition */
 #define mn tm.mt
@@ -711,6 +710,8 @@
 
     if (!fn)
 	die ("'%s'/'%s' filename extension required:  %s", BMP_BMP, BMP_CONF, filename);
+    if (!lh)
+	lh = (void*)(tm.buffer + ((intptr_t)&tm.mt.row - (intptr_t)&tm.bmlh.row));
 
     if (strcmp(fn, BMP_CONF)==0) transfer_params(filename);
     if (strcmp(fn, BMP_BMP)==0) edit_bitmap(filename);
