diff -Naur a/src/edit.c c/src/edit.c
--- a/src/edit.c	2015-11-21 21:50:17.000000000 -0200
+++ c/src/edit.c	2018-12-19 21:46:14.921350172 -0200
@@ -192,8 +192,7 @@
 	   BITMAPLILOHEADER bmlh;
 	} tm;
 static	MENUTABLE *menu = &tm.mt;
-static	BITMAPLILOHEADER *lh = (void*)(tm.buffer +
-    			((intptr_t)&tm.mt.row - (intptr_t)&tm.bmlh.row));
+static	BITMAPLILOHEADER *lh = (void*)tm.buffer;
 
 /* a convenience definition */
 #define mn tm.mt
