diff --git a/inc/config.h b/inc/config.h
index 1119760..40d7abb 100644
--- a/inc/config.h
+++ b/inc/config.h
@@ -4,8 +4,8 @@
 #define PKG_CHK "/etc/chksum"
 #define PKG_TMP "/var/pkg/cache/"
 
-#define PKG_SRC "url"
-#define PKG_SDB "url"
+#define PKG_SRC "https://downloads.sourceforge.net/project/eltanin-os/repository/glacies/x86_64/"
+#define PKG_SDB "https://downloads.sourceforge.net/project/eltanin-os/repository/glacies/"
 #define PKG_FSG "chksum"
 #define PKG_FDB "remote.db"
 #define PKG_FMT ".pkg.tzz"
