Index: scripts/build-binutils-gdb
===================================================================
--- scripts/build-binutils-gdb	(revision 7618)
+++ scripts/build-binutils-gdb	(working copy)
@@ -52,11 +52,11 @@
   HOST_CC="${HOST_CC} -static"
 fi
 
-check_library libc++.a "Disabling gdb build."
-if [ "${CROSS}" != "yes" -a "$?" != "0" ] ; then
-  # C++ isn't ready yet, don't build gdb.
-  DISABLE_GDB="--disable-gdb --disable-libdecnumber --disable-readline --disable-sim"
-fi
+#check_library libc++.a "Disabling gdb build."
+#if [ "${CROSS}" != "yes" -a "$?" != "0" ] ; then
+#  # C++ isn't ready yet, don't build gdb.
+DISABLE_GDB="--disable-gdb --disable-libdecnumber --disable-readline --disable-sim"
+#fi
 
 configure() {
   # Configuring package binutils-gdb for all targets, adding a couple.
Index: scripts/libraries
===================================================================
--- scripts/libraries	(revision 7618)
+++ scripts/libraries	(working copy)
@@ -3,16 +3,16 @@
 ## musl (libc) and compiler-rt are handled specially in the make-libraries script.
 ## Comment out any undesired libraries with a single '#'.
 ## The next three libraries are required to build binutils-gdb and LLVM/clang.
-curses
-expat
+#curses
+#expat
 libcxx
 ## The following libraries are not used for ELLCC proper, but should be built
 ## in order where libraries that depend on other libraries are built later.
-mbedtls
-c-ares
-zlib
-libssh2
-libmetalink
-nghttp2
-curl
-libedit
+#mbedtls
+#c-ares
+#zlib
+#libssh2
+#libmetalink
+#nghttp2
+#curl
+#libedit
